### PR TITLE
Fetch activation keys

### PR DESCRIPTION
**Pull request by jkozol**
Opening this as a a draft PR because I am not sure of the best testing strategy here. I load the activation keys by an async api call. In our tests we need to mock this call as I have done in the beforeAll function. Also, we need to await on the registration step to make sure the promise has resolved before continuing. So, any test that navigates to the Registration page must be async. @Gundersanne what are your thoughts on this? Do you see a better way to test this?

This commit: https://github.com/RedHatInsights/image-builder-frontend/pull/571/commits/a50fb2126816bf1e4be8d1ffda124e95a4cc4f1c shows best what I am talking about.

### CODE REVIEW

**Comment by Gundersanne** [comment]
I think it's ok if the test skips registration in all cases except the 'Step Registration' and 'with valid values' tests.

And there I'd wait until one of the options is available. In case of the 'with valid values' test I'd select an option and verify it as it's done currently by comparing the request.

There's no problem with tests being async, as jest will just await it.

**Comment by jkozol** [comment]
@Gundersanne So I don't believe we can skip the test registration step unless we switch to testing a CentOS image. We test the Wizard as a whole so have to click through the tests. Any navigation to the Registration step triggers the api call. Any reason to not by default test against CentOS?

**Comment by jkozol** [comment]
I can also just switch all the tests that require navigating past the registration page over to async but it feels a little weird to do this just to mock an api call the tests don't test.

**Comment by Gundersanne** [comment]
> @Gundersanne So I don't believe we can skip the test registration step unless we switch to testing a CentOS image. We test the Wizard as a whole so have to click through the tests. Any navigation to the Registration step triggers the api call. Any reason to not by default test against CentOS?

I'd just not wait on the api call if it's not the registration step and click the register later radio button. Or will those buttons only render once the api call has been complete?

**Comment by jkozol** [comment]
So we don't have to wait on the api call but we get a bunch of error messages about `Warning: An update to ActivationKeys inside a test was not wrapped in act(...).` I believe this is because the component updates after our test completes since the promise will return at some point. So any time we navigate to the registration page we will need to `await`. 

https://kentcdodds.com/blog/fix-the-not-wrapped-in-act-warning

**Comment by jkozol** [comment]
Screenshot from 2022-02-09 19-02-29
Screenshot from 2022-02-09 19-02-17

**Comment by jkozol** [comment]
Screenshot from 2022-02-11 13-39-38
Screenshot from 2022-02-11 13-39-18

File: src/Components/CreateImageWizard/formComponents/ActivationKeys.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+                typeAheadAriaLabel="Select activation key">
+                {isLoading &&
+                    <SelectOption isNoResultsOption={ true } data-testid='activation-keys-loading'>
+                        <Spinner isSVG size="lg" />
+                    </SelectOption>
+                }
+                {activationKeys.map((key, index) => (
+                    <SelectOption
+                        key={ index }
+                        value={ key.id }
```
Comment (Gundersanne) [inline src/Components/CreateImageWizard/formComponents/ActivationKeys.js:53]: I'd actually make the key.name the value and the description the id. Not sure the id is even exposed on access.

Comment (jkozol) [reply src/Components/CreateImageWizard/formComponents/ActivationKeys.js:53]: I was unsure about that. So if a user was to type you think they would type in the descriptive value?

Comment (teg) [reply src/Components/CreateImageWizard/formComponents/ActivationKeys.js:53]: Agreed. I don't think the id is ever used except internally by RH, right? I.e., the thing we embed in our images is the name, not the ID, so the customers would not see them? (not sure I understand how that makes sense, but that's another matter)

Comment (jkozol) [reply src/Components/CreateImageWizard/formComponents/ActivationKeys.js]: Okay cool. Changed.

Comment (jkozol) [reply src/Components/CreateImageWizard/formComponents/ActivationKeys.js]: I switched them but should we remove the id completely?

Comment (teg) [reply src/Components/CreateImageWizard/formComponents/ActivationKeys.js]: Hm... Maybe? If it is not shown anywhere else, I would. Are the names forced to be unique?

**Comment by jkozol** [comment]
Screenshot from 2022-02-11 18-55-14

File: src/Components/CreateImageWizard/formComponents/ActivationKeys.js (hunk 1 above)
Comment (jkozol) [reply src/Components/CreateImageWizard/formComponents/ActivationKeys.js]: Only displaying the names now.

Comment (jkozol) [reply src/Components/CreateImageWizard/formComponents/ActivationKeys.js]: Per this mock: https://marvelapp.com/prototype/6g7dja9/screen/76602546

File: src/constants.js [hunk 2]
```diff
@@ -1,2 +1,3 @@
 export const IMAGE_BUILDER_API = '/api/image-builder/v1';
+export const INISGHTS_API = '/api/rhsm/v2';
```
Comment (Gundersanne) [inline src/constants.js:2]: Can we rename this to RHSM_API? 'insights' is a little vague.

File: src/Components/CreateImageWizard/steps/registration.js [hunk 3]
```diff
@@ -79,7 +77,7 @@ export default {
                 {
                     type: validatorTypes.REQUIRED,
                 },
-            ],
```
Comment (Gundersanne) [inline src/Components/CreateImageWizard/steps/registration.js:82]: nitpick, trailing comma diff. I prefer with trailing commas whenever possibl.

**Review by Gundersanne — CHANGES_REQUESTED** [review]
Looking good, one last comment.

File: src/Components/CreateImageWizard/steps/registration.js (hunk 3 above)
Comment (jkozol) [reply src/Components/CreateImageWizard/steps/registration.js]: fixed

**Review by Gundersanne — APPROVED** [review]

**Pull request merged**